### DISCUSSION TITLE
Data provider filter odd behavior

### DISCUSSION THREAD

**Discussion by FrancisksJemeljanovs** [Budibase/budibase#10515] [Help] [answered]
Hi!

I have a data provider that is using Postgres SQL table as source. I want to do the filtering but this one column is having some weird behavior. The column in question is of type "Varchar" in Postgres and "text" in Budibase.

Currently table contains all NULL values in this column, but most of them will get populated. I need set a filter to filter out specific string values when they will get populated, so I hoped to just use Not equals - value - "necessary value".

Current behavior I am getting is:
- putting 'Not equals' -> value -> any text value -  is giving me empty table (?)
- putting 'Equals' -> value -> any text value - is giving me empty table
- putting 'Not equals' -> value -> leave field empty (empty string) - is giving me full table with all records (?)
- putting 'Equals' -> value -> leave field empty (empty string) - is giving me full table with all records as expected
- putting 'Is empty' - is giving me full table with all records as expected
- putting 'Is not empty' - is giving me empty table as expected
- putting 'Equals'/'Not equals' -> binding -> JS function: return null; - is giving full table regardless of putting filter to Equals or Not equals. (?)

I had this exact filtering logic working in my previous app version where this column was Enum in PostgreSQL side and never had NULL values. There are external reasons why this column cannot be Enum anymore. I have done re-fetching the table in Budibase after column type changes in DB side.

**Comment by melohagan** [comment]
Hey @FrancisksJemeljanovs 

I appreciate the detailed scenarios here. Could you also provide me with the schema of your table (create script). Then I can investigate further.

**Reply by FrancisksJemeljanovs** [reply, marked as answer]
Hi @melohagan! I have gotten smarter since I first started this discussion. There are no problems with Budibase Data provider filter behavior.

Somehow my basic SQL knowledge failed me on this one. Turns out I had never encountered such scenario where I have to exclude some specific values in a list of values that also include nulls.

Sorry for inconvenience!

**Reply by melohagan** [reply]
No worries at all @FrancisksJemeljanovs 

Thanks for letting me know!
